### PR TITLE
Fix transaction serialization

### PR DESCRIPTION
**Pull request by aslesarenko**
This PR fixes the tx serialization regression bug.

### CODE REVIEW

File: core/shared/src/main/scala/sigma/util/CollectionUtil.scala [hunk 1]
```diff
@@ -41,6 +41,18 @@ object CollectionUtil {
     result
   }
 
+  /** Concatenate all the arrays in the sequence. */
+  def concatArrays[T: ClassTag](seq: Traversable[Array[T]]): Array[T] = {
```
Comment (kushti) [inline core/shared/src/main/scala/sigma/util/CollectionUtil.scala:45]: Adding here method for one test just is overkill, for tests ++ is okay

File: interpreter/shared/src/main/scala/sigmastate/interpreter/InterpreterContext.scala [hunk 2]
```diff
@@ -48,13 +45,17 @@ object ContextExtension {
...
-        .map(_ => (r.getByte(), r.getValue().asInstanceOf[EvaluatedValue[_ <: SType]]))
-      ContextExtension(mutable.LinkedHashMap(ext:_*))
-    }
 
-  }
+      val values = (0 until extSize)
+          .map(_ => (r.getByte(), r.getValue().asInstanceOf[EvaluatedValue[_ <: SType]]))
 
+      // TODO v6.0: enforce ordering e.g. using the code below (see https://github.com/ScorexFoundation/sigmastate-interpreter/issues/681)
+      // if (VersionContext.current.activatedVersion > JitActivationVersion) {
```
Comment (kushti) [inline interpreter/shared/src/main/scala/sigmastate/interpreter/InterpreterContext.scala:53]: Such Versioning does not work here, as old clients will be broken after activation. Remove misleading TODO

File: sdk/shared/src/main/scala/org/ergoplatform/sdk/JsonCodecs.scala [hunk 3]
```diff
@@ -245,7 +245,13 @@ trait JsonCodecs {
 
   implicit val contextExtensionDecoder: Decoder[ContextExtension] = Decoder.instance({ cursor =>
     for {
-      values <- cursor.as[mutable.LinkedHashMap[Byte, EvaluatedValue[SType]]]
+      values <- {
+        // TODO v6.0: enforce ordering e.g. using the code below (see https://github.com/ScorexFoundation/sigmastate-interpreter/issues/681)
+        // if (VersionContext.current.activatedVersion > JitActivationVersion) {
+        //   ContextExtension(mutable.LinkedHashMap(values: _*))
```
Comment (kushti) [inline sdk/shared/src/main/scala/org/ergoplatform/sdk/JsonCodecs.scala:251]: the same TODO issue basically

**Review by kushti — CHANGES_REQUESTED** [review]

File: core/shared/src/main/scala/sigma/util/CollectionUtil.scala (hunk 1 above)
Comment (aslesarenko) [reply core/shared/src/main/scala/sigma/util/CollectionUtil.scala:45]: removed

File: interpreter/shared/src/main/scala/sigmastate/interpreter/InterpreterContext.scala (hunk 2 above)
Comment (aslesarenko) [reply interpreter/shared/src/main/scala/sigmastate/interpreter/InterpreterContext.scala:53]: removed

File: sdk/shared/src/main/scala/org/ergoplatform/sdk/JsonCodecs.scala (hunk 3 above)
Comment (aslesarenko) [reply sdk/shared/src/main/scala/org/ergoplatform/sdk/JsonCodecs.scala:251]: removed

**Comment by aslesarenko** [comment]
@kushti what should I do with failing serialization roundtrip tests?

**Comment by kushti** [comment]
> @kushti what should I do with failing serialization roundtrip tests?

I think an issue, to start with

**Comment by aslesarenko** [comment]
> I think an issue, to start with

There is an issue https://github.com/ScorexFoundation/sigmastate-interpreter/issues/681
Should I refer to it from the ignored tests?

File: core/shared/src/main/scala/sigma/util/CollectionUtil.scala (hunk 1 above)
Comment (kushti) [reply core/shared/src/main/scala/sigma/util/CollectionUtil.scala:45]: push the code

**Comment by kushti** [comment]
> > I think an issue, to start with
> 
> There is an issue #681 Should I refer to it from the ignored tests?

I guess vice-versa , so you should refer to the tests from the issue, which is about way different level of visibility. 

The issue is far from ideal , as talks there are mostly about JSON serialization

**Comment by aslesarenko** [comment]
> talks there are mostly about JSON serialization

Actually, the problem is not just with binary, but also with Json, and issue highlights which complexity. 
Will add references.

**Review by kushti — APPROVED** [review]

**Pull request merged**